### DISCUSSION TITLE
Pass RM code into json file

### DISCUSSION THREAD

**Discussion by massimocapoccia** [bordaigorl/rmview#49] [General]
Hi, i customized the json file to pass the IP and pwd into the file for auto login, but i am getting an error xxxxx "expected unicode ...". Any help? thanks

**Comment by ndehaan** [comment]
I have the same issue.  The password field is expecting unicode or bytes.

Config file contents:

{
    "ssh": {
        "address": 	["10.0.1.37"],
	"auth_method":	["password"],
	"password":	["password"]
    },
    "orientation": "portrait",
    "pen_size": 15,
    "pen_color": "blue",
    "pen_trail": 200
}

![Screenshot 2021-05-17 181546](https://user-images.githubusercontent.com/3884715/118575759-0bdebe80-b73c-11eb-85a1-054b57767e7d.png)

**Reply by bordaigorl** [reply]
Your config file is malformed.
THe square brackets create a list. The `address` field can be either a string or a list (when it is a list the app will allow you to pick which you want to connect to at startup, which is useful for multi-device or multi-connection setups).
The `auth_method` and `password` fields, however cannot be lists:

```
{
"ssh": {
"address": ["10.0.1.37"],
"auth_method": "password",
"password": "YOUR_ACTUAL_PASSWORD"
},
"orientation": "portrait",
"pen_size": 15,
"pen_color": "blue",
"pen_trail": 200
}
```

If you are connecting only using that address, it may just be better to set `"address": "10.0.1.37"`.

**Reply by ndehaan** [reply]
Thank-you, this now works for me.

**Comment by LenaHannah** [comment]
thank you so much for creating this, it looks great!
i am just trying to connect, and not sure, where do I find the password that goes into "YOUR_ACTUAL_PASSWORD"?

**Reply by bordaigorl** [reply]
Connect your reMarkable to Wi-Fi or USB. You will find your SSH password and IP address in the settings menu:
<kbd>Settings > Help > Copyrights and licenses > General information</kbd> (scroll down)
